### ISSUE TITLE
Android - Intent with extra parameters to launch games from Front Ends. 

### ISSUE THREAD

**Issue by flatmapthatshit** [stenzek/duckstation#1847]
We need a way to allow android front-ends like DIG, Arc Browser etc to launch games in DuckStation. Games should autostart in full screen; closing the game should exit DuckStation 

**Suggested solutions**:

The standard practice is to have a launch intention that takes the path to the game. Some emulators implement this by reusing the `MainActivity` and adjusting it to handle a custom intent with extra parameters. Other emulators implement a separate activity to handle the launch intent.
Here's a list of emulators and launch intents: https://hyperspin-fe.com/forums/topic/8537-settings-android-emulator-settings/

Basically, if you can launch a game correctly using `adb shell start` then front-ends can launch DuckStation.

If a launch intent already exists, it would be good to document it.

Also, as a bonus, it would be great if the launch intent could accept optional parameters to adjust settings on a per-session basis. E.g., 4:3 vs 16:9, upscaling settings, enabling and disabling wide-screen hacks, PGXP, etc. If you ever decide to implement shaders and filters it would also be great to be able to control it on a per session basis (e.g., to set custom bezels, etc).

**Comment by stenzek** [comment]
https://github.com/stenzek/duckstation/blob/a5e316b91651636ce6d701d55160fe507cf35a80/android/app/src/main/java/com/github/stenzek/duckstation/MainActivity.java#L396-L399

No plans to add further launch options. It's not trivial to do, and you can set everything per game anyway.
